packages/dev/core/src/Meshes/trailMesh.ts

```diff
@@ -228,6 +228,38 @@ export class TrailMesh extends Mesh {
         }
     }
 
+    /**
+     * Reset trailing mesh geometry.
+     */
+    public reset(): void {
+        const positions = this.getVerticesData(VertexBuffer.PositionKind);
+        const normals = this.getVerticesData(VertexBuffer.NormalKind);
+        const wm = this._generator.getWorldMatrix();
+        if (positions && normals) {
+            const alpha: number = (2 * Math.PI) / this._sectionPolygonPointsCount;
+            for (let i: number = 0; i <= this._sectionPolygonPointsCount; i++) {
+                const angle = i !== this._sectionPolygonPointsCount ? i * alpha : 0;
+                this._sectionVectors[i].copyFromFloats(Math.cos(angle) * this.diameter, Math.sin(angle) * this.diameter, 0);
+                this._sectionNormalVectors[i].copyFromFloats(Math.cos(angle), Math.sin(angle), 0);
+                Vector3.TransformCoordinatesToRef(this._sectionVectors[i], wm, this._sectionVectors[i]);
+                Vector3.TransformNormalToRef(this._sectionNormalVectors[i], wm, this._sectionNormalVectors[i]);

```

Comment:
just wondering - this is (unless I missed something) a direct copy from the update function. Can't we reuse the code in update? Extract the repeated code?
Or maybe add a "resetOnNextFrame" flag instead of adding a function

Reply:
yh, I was considering to re-factor the code since updating the first section seems to be a common use case. It would have to be for the entire loop (including `alpha`). Did not do it because not sure if this reset strategy is robust enuff. For instance, if there is a huge trail diameter, should the reset still put all points along the circle? Or will it be better to coalesce/collapse the trail to a line/point? What do you think?

> Or maybe add a "resetOnNextFrame" flag instead of adding a function

sorry, erm, I do not understand the question here. You are suggesting adding a "resetOnNextFrame" boolean in the update function to allow the trail to reset in between updates? If so, as a user, I do not want to have the trail update running every frame when I know the generator isn't moving. I guess it boils down to: which is more expensive? adding/removing observer when needed or multiple `getVerticesData, for loops` every frame? The PG is a fairly extreme repeating case, tho...

Reply:
I get the 2nd point - what you are saying is that a trail mesh that hasn't started will not be able to call "reset". Fair point, ignore my suggestion :-)

To your first point - I ran this - https://playground.babylonjs.com/?snapshot=refs/pull/15369/merge#1F4UET#58 . On click the trail is being reset. It seems to be working as expected. It's a good question, but i'll have to play around with it a bit more to have a good answer. I'll ask the team for some feedback as well.

Reply:
Let's keep the solution as is, just reuse as much code as possible.

Reply:
Done! Did a small refactor to re-use code. Thanks!